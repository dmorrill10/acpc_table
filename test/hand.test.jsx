import React from 'react';
import renderer from 'react-test-renderer';
const Hand = require('../dist/hand.js');

test('Two face-down card hand works', () => {
  const cards = [{}, {}];
  const patient = <Hand cards={cards}/>;
  const tree = renderer.create(patient).toJSON();
  expect(tree).toMatchSnapshot();
});
test('Two face-up card hand works', () => {
  const cards = [
    {
      rank: '2',
      suit: '&hearts;'
    }, {
      rank: '3',
      suit: '&diamonds;'
    }
  ];
  const patient = <Hand cards={cards}/>;
  const tree = renderer.create(patient).toJSON();
  expect(tree).toMatchSnapshot();
});
