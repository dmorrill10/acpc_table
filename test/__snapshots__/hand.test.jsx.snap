exports[`test Two face-down card hand works 1`] = `
<ol
  className="hand">
  <li>
    <div
      className="card-face-down">
      <div
        className="logo" />
    </div>
  </li>
  <li>
    <div
      className="card-face-down">
      <div
        className="logo" />
    </div>
  </li>
</ol>
`;

exports[`test Two face-up card hand works 1`] = `
<ol
  className="hand">
  <li>
    <div
      className="card-face-up">
      <div
        className="top_left_rank">
        2
      </div>
      <div
        className="suit">
        &hearts;
      </div>
      <div
        className="bottom_right_rank">
        2
      </div>
    </div>
  </li>
  <li>
    <div
      className="card-face-up">
      <div
        className="top_left_rank">
        3
      </div>
      <div
        className="suit">
        &diamonds;
      </div>
      <div
        className="bottom_right_rank">
        3
      </div>
    </div>
  </li>
</ol>
`;
